### PR TITLE
use shared frontend library

### PR DESCRIPTION
**Pull request by vladimir-8**
refactoring:

- code that shows a colored line near linked tasks is moved into the [shared library ](https://github.com/coopcycle/coopcycle-frontend-js)to be re-used in the mobile app https://github.com/coopcycle/coopcycle-app/pull/788

- some dispatch selectors are moved and used from the [shared library ](https://github.com/coopcycle/coopcycle-frontend-js)

Screen Shot 2020-09-03 at 17 41 18

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
   "dependencies": {
     "@mapbox/polyline": "^1.1.0",
     "co": "^4.6.0",
+    "coopcycle-frontend-js": "git+https://git@github.com/coopcycle/coopcycle-frontend-js.git#1.0.4",
```
Comment (alexsegura) [inline package.json:22]: Frontend dependencies should be in `devDependencies`

File: package.json [hunk 2]
```diff
@@ -34,6 +35,7 @@
     "pg-hstore": "^2.3.3",
     "promise": "^8.1.0",
     "redis": "^3.0",
+    "reduce-reducers": "^1.0.4",
```
Comment (alexsegura) [inline package.json:38]: Frontend dependencies should be in devDependencies

**Comment by vladimir-8** [comment]
@alexsegura I think I found a place where I have not switched to the new selectors (in `actions.js`), I'll fix it soon

**Comment by alexsegura** [comment]
@vladimir-8 can you also check why tests are failing please?

**Comment by vladimir-8** [comment]
I'll do, may be these issues are related

**Comment by vladimir-8** [comment]
@alexsegura both problems are fixed now

**Comment by alexsegura** [comment]
I rebased to fix conflicts, testing it now 🙂

**Review by alexsegura — APPROVED** [review]

**Comment by alexsegura** [comment]
Thank you @vladimir-8

**Pull request merged**